brian2/tests/test_spikegenerator.py

```diff
@@ -102,6 +102,34 @@ def test_spikegenerator_period():
         assert_allclose(generator_spikes, recorded_spikes)
 
 
+@attr('codegen-independent')
+def test_spikegenerator_extreme_period():
+    '''
+    Basic test for `SpikeGeneratorGroup`.
+    '''
+    indices = np.array([0, 1, 2])
+    times   = np.array([0, 1, 2]) * ms
+    SG = SpikeGeneratorGroup(5, indices, times, period=1e6*second)
+    s_mon = SpikeMonitor(SG)
+    with catch_logs() as l:
+        run(10*ms)
+
+    assert_equal(s_mon.i, np.array([0, 1, 2]))
+    assert_allclose(s_mon.t, [0, 1, 2]*ms)
+    assert len(l) == 1 and l[0][1].endswith('spikegenerator_long_period')
+    print(l)
+
+@attr('standalone-compatible')
+@with_setup(teardown=reinit_devices)
+def test_spikegenerator_period_rounding():
+    # See discussion in PR #1042
+    src = SpikeGeneratorGroup(1, [0, 0, 0], [0*ms, .9*ms, .99999*ms],
+                              period=1*ms)
+    mon = SpikeMonitor(src)
+    run(2 * ms)
+    assert_allclose(mon.t, [0., 0.9, 1., 1.9] *msecond)
+
```

Comment:
This test is wrong. Actually, this is an error because the spike at .99999ms is not emitted. So we can keep this test but it ought to raise an error. See the error in logic above.

Reply:
Oops, I added this a bit hastily and did somewhat miss the point of this example... It's a bit weird for a test since it tests for expected behaviour that is certainly not expected for the user. But still, definitely better than silently ignoring the spike.